### ISSUE TITLE
Support for tags on azurerm_network_watcher_flow_log

### ISSUE THREAD

**Issue by joedoss** [terraform-providers/terraform-provider-azurerm#11442]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

azurerm_network_watcher_flow_log should allow a "tags" parameter, since the underlying resource supports tagging.
(Microsoft.Network/networkWatchers/flowLogs)

### New or Affected Resource(s)

Affected resource: 
azurerm_network_watcher_flow_log

### Potential Terraform Configuration

```hcl
resource "azurerm_network_watcher_flow_log" "exampleFlowlog" {
  network_watcher_name = azurerm_network_watcher.networkwatcher.name
  resource_group_name  = azurerm_resource_group.networkwatcherrg.name

  network_security_group_id = azurerm_network_security_group.ManagementNSG.id
  storage_account_id        = azurerm_storage_account.nsgflowlogstorage.id
  enabled                   = true
  version                   = 2
  tags = {
    "deployment-environment"   = "poc",
    "deployment-method"        = "terraform",
    "business-applicationname" = "infrastructure"
  }

  retention_policy {
    enabled = true
    days    = 120
  }

  traffic_analytics {
    enabled               = true
    interval_in_minutes   = 10
    workspace_id          = azurerm_log_analytics_workspace.hub-loganalyticsworkspace.workspace_id
    workspace_region      = azurerm_log_analytics_workspace.hub-loganalyticsworkspace.location
    workspace_resource_id = azurerm_log_analytics_workspace.hub-loganalyticsworkspace.id
  }
}
```

### References

https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/tag-support#microsoftnetwork

* #0000

**Comment by MitchDrage** [comment]
Not supported on the current network SDK version. Need to bump SDK from 2020-05-01 to a minimum of 2020-06-01 to get support for tags on that resource type.

**Comment by katbyte** [comment]
looks like this was fixed by #11670
